sdk/core/AzureCore/Tests/QueryParametersTests.swift

```diff
@@ -64,14 +64,28 @@ enum ShapeEnum: QueryStringConvertible {
     }
 }
 
-typealias MyDate = Date
-extension MyDate: QueryStringConvertible {
-    internal static let dateFormat = "yyyy-MM-dd"
+final class MyDate: DateConvertible {
+    var date: Date
 
-    public var queryValue: String {
+    public static let dateFormat = "yyyy-MM-dd"
+
+    internal static var formatter: DateFormatter = {
         let formatter = DateFormatter()
-        formatter.dateFormat = MyDate.dateFormat
-        return formatter.string(from: self)
+        formatter.dateFormat = dateFormat
+        return formatter
+    }()
+
+    init?(string: String) {
+        guard let date = MyDate.formatter.date(from: string) else { return nil }
+        self.date = date
+    }
+
+    init(_ date: Date) {
+        self.date = date
+    }
+
+    public var queryValue: String {
+        return MyDate.formatter.string(from: date)
     }
 }
```

Comment:
@sacheu @bsiegel this is an example of how we can handle dates in AzureCore instead of in the code gen itself. We could have a `Dates.swift` file with these generated the same way we have an `Enumerations.swift` file.